runelite-client/src/main/java/net/runelite/client/plugins/puzzlesolver/PuzzleSolver.java

```diff
@@ -35,312 +23,210 @@
  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-package net.runelite.client.plugins.puzzlesolver;
 
-import java.util.ArrayList;
-import java.util.List;
+package net.runelite.client.plugins.puzzlesolver;
 
-/**
- * This 5x5 sliding puzzle solver algorithm is based on https://www.jaapsch.net/puzzles/fifteen.htm.
- */
-public class PuzzleSolver
+public class PuzzleSolver implements Runnable
 {
 	public static final int BLANK_TILE_VALUE = -1;
 	public static final int DIMENSION = 5;
-	private static final int SIZE = DIMENSION * DIMENSION - 1;
+	public static final int SIZE = DIMENSION * DIMENSION;
+
+	private static final int MAX_DEPTH = 64;
```

Comment:
Are you 110% sure that puzzles always have less than 65 steps to the solution?

Reply:
No, but after probably 200+ puzzles my longest one was 54 steps and like 95% of them are < 40 steps.
It is a bit of a guess because i know they love powers of two and it seemed reasonable.
The solver also starts getting a bit slow around 60 steps.